src/ngx_http_lua_headers_out.c

```diff
@@ -319,62 +319,40 @@ ngx_http_set_builtin_multi_header(ngx_http_request_t *r,
 
     headers = (ngx_table_elt_t **) ((char *) &r->headers_out + hv->offset);
 
-    if (hv->no_override) {
-        for (h = *headers; h; h = h->next) {
-            if (!h->hash) {
-                h->value = *value;
-                h->hash = hv->hash;
-                return NGX_OK;
-            }
-        }
-
-        goto create;
-    }
-
-    /* override old values (if any) */
-
-    if (*headers) {
-        for (h = (*headers)->next; h; h = h->next) {
+    for (h = *headers; h; h = h->next) {
+        if (!hv->no_override) {
             h->hash = 0;
             h->value.len = 0;
         }
+    }
 
-        h = *headers;
-
+    if (h->hash == 0) {
+        // update the last element if possible
         h->value = *value;
-
-        if (value->len == 0) {
-            h->hash = 0;
-
-        } else {
+        if (value->len != 0) {
             h->hash = hv->hash;
         }
 
-        return NGX_OK;
-    }
-
-create:
-
-    for (ph = headers; *ph; ph = &(*ph)->next) { /* void */ }
+    } else {
+        ho = ngx_list_push(&r->headers_out.headers);
```

Comment:
when last element is valid and other elements are not, here will make a new element, right? Does it mean that the invalid elements never be reused in pool? So why not just nullifying the first element `next` for simplicity?

Reply:
Actually when set builtin multi input header, it just nullifying the next pointer.

can we do same thing for output headers?